### PR TITLE
NMS-13340: Add HostResourceSWRunDetector docs

### PR DESCRIPTION
**Pull request by mfuhrmann**
### All Contributors

* [x] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [x] Have you made a comment in that issue which points back to this PR?
* [x] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [x] If this is a new or updated feature, is there documentation for the new behavior?
* [ ] If this is new code, are there unit and/or integration tests?
* [ ] If this PR targets a `foundation-*` branch, does it avoid changing files in `$OPENNMS_HOME/etc/`?

### External References

* JIRA (Issue Tracker): https://issues.opennms.org/browse/NMS-13340

### CODE REVIEW

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+==== Detector facts
+
+[options="autowidth"]
+|===
+| Implementation | `org.opennms.netmgt.provision.detector.snmp.HostResourceSWRunDetector`
+|===
+
+==== Configuration and Use
+
+.Parameters for the <DETECTOR-NAME-HERE>
```
Comment (indigo423) [inline docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:24]: <DETECTOR-NAME-HERE>

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc [hunk 2]
```diff
@@ -0,0 +1,39 @@
+=== HostResourceSWRunDetector
+
+This detector test the running state of one or more processes.
+It does this via SNMP by inspecting the _hrSwRunTable_ of the http://www.ietf.org/rfc/rfc2790[HOST-RESOURCES-MIB].
```
Comment (indigo423) [inline docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:4]:
```suggestion
Detect a process for monitoring using the http://www.ietf.org/rfc/rfc2790[HOST-RESOURCES-MIB].
```

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc [hunk 3]
```diff
@@ -0,0 +1,39 @@
+=== HostResourceSWRunDetector
+
+This detector test the running state of one or more processes.
```
Comment (indigo423) [inline docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:3]:
```suggestion
```

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+It does this via SNMP by inspecting the _hrSwRunTable_ of the http://www.ietf.org/rfc/rfc2790[HOST-RESOURCES-MIB].
+The test is done by matching a given process as _hrSwRunName_ against the numeric value of the _hrSwRunState_.
```
Comment (indigo423) [inline docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:5]:
```suggestion
A process is detected by a given process as _hrSwRunName_ matching against the numeric value of the _hrSwRunState_.
```

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+This detector uses _SNMP_ to accomplish its work.
```
Comment (indigo423) [inline docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:7]:
```suggestion
The SNMP protocol needs to be configured and the HOST-RESOURCES-MIB must be supported by the SNMP agent.
```

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+Therefore systems against which it is to be used must have an SNMP agent installed and configured.
```
Comment (indigo423) [inline docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:8]:
```suggestion
```

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+Furthermore, the _SNMP agent_ on the system must support the _HOST-RESOURCES-MIB_.
```
Comment (indigo423) [inline docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:9]:
```suggestion
```

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+Most modern _SNMP agents_, including most distributions of the _Net-SNMP agent_ and the _SNMP service_ that ships with _Microsoft Windows_, support this _MIB_.
```
Comment (indigo423) [inline docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:10]:
```suggestion
Most modern _SNMP agents_, including most distributions of the _Net-SNMP agent_ and the _SNMP service_ that ships with _Microsoft Windows_, support this _MIB_.
```
```suggestion
Most modern _SNMP agents_, e.g. _Net-SNMP agent_ and the _Microsoft SNMP service_ support the _HOST-RESOURCES-MIB_.
```

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+Out-of-box support for _HOST-RESOURCES-MIB_ among commercial _Unix_ operating systems may be somewhat spotty.
```
Comment (indigo423) [inline docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:11]:
```suggestion
```

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+NOTE: This detector implements the <<provisioning/detectors/SnmpDetector.adoc>>.
```
Comment (indigo423) [inline docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:13]:
```suggestion
NOTE: This detector implements the configuration parameters inherited from the <<provisioning/detectors/SnmpDetector.adoc>>.
```

**Review by indigo423 — COMMENTED** [review]
Some minor bla bla changes, I guess @Bonrob2 as a native speaker can add some 🌸 as well :)

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc (hunk 1 above)
Comment (Bonrob2) [reply docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:24]: We should replace everything with blah blah! :-)

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc [hunk 11]
```diff
@@ -0,0 +1,40 @@
+=== HostResourceSWRunDetector
+
+This detector test the running state of one or more processes.
+It does this via SNMP by inspecting the _hrSwRunTable_ of the http://www.ietf.org/rfc/rfc2790[HOST-RESOURCES-MIB].
```
Comment (Bonrob2) [inline docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:4]:
```suggestion
Use the http://www.ietf.org/rfc/rfc2790[HOST-RESOURCES-MIB] to detect a process to monitor.
```
Is the process doing the monitoring, or are we detecting a process to monitor?

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+The test is done by matching a given process as _hrSwRunName_ against the numeric value of the _hrSwRunState_.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:5]:
```suggestion
A given process matches against the numeric value of the _hrSwRunState_ to detect a process as _hrSwRunName_. 
```

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+
+This detector uses _SNMP_ to accomplish its work.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:7]:
```suggestion
You must configure the SNMP protocol, and the SNMP agent must support HOST-RESOURCES-MIB.
```

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc (hunk 5 above)
Comment (Bonrob2) [reply docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:7]: updates to Ronny's change below.

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc (hunk 4 above)
Comment (Bonrob2) [reply docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:5]: update below

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+Therefore systems against which it is to be used must have an SNMP agent installed and configured.
+Furthermore, the _SNMP agent_ on the system must support the _HOST-RESOURCES-MIB_.
+Most modern _SNMP agents_, including most distributions of the _Net-SNMP agent_ and the _SNMP service_ that ships with _Microsoft Windows_, support this _MIB_.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:10]: update to comment below.

Comment (Bonrob2) [inline docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:10]:
```suggestion
Most modern _SNMP agents_, e.g., _Net-SNMP agent_ and the _Microsoft SNMP service_, support the _HOST-RESOURCES-MIB_.
```

**Review by Bonrob2 — CHANGES_REQUESTED** [review]

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc (hunk 1 above)
Comment (mfuhrmann) [reply docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:22]: https://i.gifer.com/1lSu.gif

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc (hunk 11 above)
Comment (mfuhrmann) [reply docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc]: We detect a running process to be able to monitor it with [HostResourceSwRunMonitor](https://github.com/OpenNMS/opennms/blob/develop/docs/modules/operation/pages/service-assurance/monitors/HostResourceSwRunMonitor.adoc).
The process is listed in the HOST-RESOURCES-MIB.

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc (hunk 1 above)
Comment (mfuhrmann) [reply docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:20]:
```suggestion
.Parameters for the HostResourceSwRunDetector
```

**Comment by mfuhrmann** [comment]
@Bonrob2 @indigo423 Thanks a lot! I hope I've applied your changes in the correct order :joy:. Please review again.

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc (hunk 11 above)
Comment (Bonrob2) [reply docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc]: Thanks. Then my suggestion is phrased correctly.

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc [hunk 15]
```diff
@@ -1,16 +1,12 @@
...
+Detect a process for monitoring using the http://www.ietf.org/rfc/rfc2790[HOST-RESOURCES-MIB].
+A given process matches against the numeric value of the _hrSwRunState_ to detect a process as _hrSwRunName_. 
 
-This detector uses _SNMP_ to accomplish its work.
-Therefore systems against which it is to be used must have an SNMP agent installed and configured.
-Furthermore, the _SNMP agent_ on the system must support the _HOST-RESOURCES-MIB_.
-Most modern _SNMP agents_, including most distributions of the _Net-SNMP agent_ and the _SNMP service_ that ships with _Microsoft Windows_, support this _MIB_.
-Out-of-box support for _HOST-RESOURCES-MIB_ among commercial _Unix_ operating systems may be somewhat spotty.
+You must configure the SNMP protocol, and the SNMP agent must support HOST-RESOURCES-MIB.
+Most modern _SNMP agents_, e.g. _Net-SNMP agent_ and the _Microsoft SNMP service_ support the _HOST-RESOURCES-MIB_.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:7]:
```suggestion
Most modern _SNMP agents_, e.g. _Net-SNMP agent_ and the _Microsoft SNMP service_, support the _HOST-RESOURCES-MIB_.
```

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
just a comma. :-)

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc [hunk 16]
```diff
@@ -1,7 +1,7 @@
 === HostResourceSWRunDetector
 
 This detector test the running state of one or more processes.
-It does this via SNMP by inspecting the _hrSwRunTable_ of the http://www.ietf.org/rfc/rfc2790[HOST-RESOURCES-MIB].
+Detect a process for monitoring using the http://www.ietf.org/rfc/rfc2790[HOST-RESOURCES-MIB].
```
Comment (Bonrob2) [inline docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:3]: Use the http://www.ietf.org/rfc/rfc2790[HOST-RESOURCES-MIB] to detect a process to monitor.

Comment (Bonrob2) [inline docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:3]:
```suggestion
Use the http://www.ietf.org/rfc/rfc2790[HOST-RESOURCES-MIB] to detect a process to monitor.
```

**Review by Bonrob2 — COMMENTED** [review]
LGTM!

**Review by Bonrob2 — COMMENTED** [review]
LGTM

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc (hunk 11 above)
Comment (mfuhrmann) [inline docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc]:
```suggestion
Use the http://www.ietf.org/rfc/rfc2790[HOST-RESOURCES-MIB] to detect a process to monitor.
```

File: docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc [hunk 17]
```diff
@@ -0,0 +1,35 @@
+=== HostResourceSWRunDetector
+
+Use the http://www.ietf.org/rfc/rfc2790[HOST-RESOURCES-MIB] to detect a process to monitor.
+A given process matches against the numeric value of the _hrSwRunState_ to detect a process as _hrSwRunName_.
```
Comment (mfuhrmann) [inline docs/modules/operation/pages/provisioning/detectors/HostResourceSWRunDetector.adoc:4]:
```suggestion
Use the http://www.ietf.org/rfc/rfc2790[HOST-RESOURCES-MIB] to detect a process to monitor.
```

**Review by Bonrob2 — APPROVED** [review]

**Pull request merged**